compiler/rustc_middle/src/mir/traversal.rs

```diff
@@ -178,7 +178,7 @@ impl<'a, 'tcx> Postorder<'a, 'tcx> {
         // When we yield `C` and call `traverse_successor`, we push `B` to the stack, but
         // since we've already visited `E`, that child isn't added to the stack. The last
         // two iterations yield `B` and finally `A` for a final traversal of [E, D, C, B, A]
-        while let Some(&mut (_, ref mut iter)) = self.visit_stack.last_mut() && let Some(bb) = iter.next_back() {
+        while let Some(bb) = self.visit_stack.last_mut().and_then(|(_, iter)| iter.next_back()) {
```

Comment:
Should this use a let chain ?

Reply:
I can drop 0e0dc59acbdc4671c1fdcaf6c06fd9c4d1d8c9e2 if you want :P

My point for using `and_then` was to show that `iter` is only used for `next_back` and is not used in the body of the loop. IMO with let chain it was a bit harder to read